replybot/lib/typewheels/machine.js

```diff
@@ -520,7 +529,13 @@ function _gatherResponses(ctx, qa, q, previous = []) {
   return [...previous, q]
 }
 
-function _response (ctx, qa, {question, validation, response, token, followUp}) {
+function _response (ctx, qa, {question, validation, response, token, followUp, isOff}) {
+  if (isOff) {
```

Comment:
Hrm, the `isOff` is a boolean that causes the execution to immediately enter a block that returns... It shares no logic with the rest of the function. Isn't this easier to make a separate function? What's the advantage of sharing the function?

Reply:
In my mind, the `_response()` method is in charge of returning, well a response. If we were to move the response to an `off-time` event, then, shouldn't this mean that we will have two methods `_response()` and `aNewMethod()` to _deal_ with responses?

Reply:
Hrm. So we're adding a 4th action to the `act` function. The other three actions all use the `respond` function and there is very little logic inside the action block, all the smarts goes inside the `respond` function. That's definitely how it was, so I think it's valid what you say, might as well leave it that way unless we refactor the whole thing. OK!